packages/react-router/src/transformer.ts

```diff
@@ -80,11 +80,11 @@ export const defaultTransformer: RouterTransformer = {
   },
 }
 
-const createTransformer = <T extends string>(
-  key: T,
-  check: (value: any) => boolean,
-  toValue: (value: any) => any = (v) => v,
-  fromValue: (value: any) => any = (v) => v,
+const createTransformer = <TKey extends string, TFrom, TTo>(
+  key: TKey,
+  check: (value: any) => value is TFrom,
+  toValue: (value: TFrom) => TTo = (v) => v as never,
```

Comment:
these being optional seems odd - they're used in all of the current transformers, and I can't think of a situation where i'd want one and not the other, or neither

Reply:
cc @chorobin @schiller-manuel

Reply:
Can you please choose different names than `TFrom` and `TTo` ? those are usually used in the navigation API for the `from` / `to` props

Reply:
renamed